File: apps/api/app/schemas/cockpit_eod_report.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class CockpitEodSummarySchema(BaseModel):
    headline: str
    opened_today: int
    closed_today: int
    open_positions_now: int
    alerts_needing_attention: int
    lessons_available: int


class CockpitEodPnlSchema(BaseModel):
    realized_day: float | None
    unrealized_snapshot: float | None
    realized_basis: str
    unrealized_basis: str


class CockpitEodOpenPositionItemSchema(BaseModel):
    asset_symbol: str
    side: str
    qty: float | None
    opened_at: str | None
    unrealized_pnl: float | None


class CockpitEodTradeItemSchema(BaseModel):
    asset_symbol: str
    side: str
    opened_at: str | None
    closed_at: str | None
    realized_pnl: float | None
    close_reason: str | None


class CockpitEodClosedPositionsSchema(BaseModel):
    count: int
    wins: int | None
    losses: int | None
    flat: int | None
    unknown: int
    best_trade: CockpitEodTradeItemSchema | None
    worst_trade: CockpitEodTradeItemSchema | None
    items: list[CockpitEodTradeItemSchema]


class CockpitEodPaperActivitySchema(BaseModel):
    opened_today: int
    closed_today: int
    current_open_positions: int


class CockpitEodIncidentItemSchema(BaseModel):
    severity: str
    code: str
    title: str
    source: str
    created_at: str | None
    detail: str | None


class CockpitEodMonitorNoteSchema(BaseModel):
    title: str
    detail: str
    severity: str
    created_at: str | None


class CockpitEodLessonSchema(BaseModel):
    title: str
    detail: str
    evidence_count: int


class CockpitEodOpenPositionsSchema(BaseModel):
    count: int
    items: list[CockpitEodOpenPositionItemSchema]


class CockpitEodReportResponseSchema(BaseModel):
    report_date: str
    generated_at: str
    mode: Literal["paper"]
    summary: CockpitEodSummarySchema
    paper_activity: CockpitEodPaperActivitySchema
    pnl: CockpitEodPnlSchema
    open_positions: CockpitEodOpenPositionsSchema
    closed_positions: CockpitEodClosedPositionsSchema
    alerts_or_incidents: list[CockpitEodIncidentItemSchema]
    monitor_notes: list[CockpitEodMonitorNoteSchema]
    lessons: list[CockpitEodLessonSchema]
    recommended_actions: list[str]
    limitations: list[str]

File: apps/web/lib/api/cockpitEodReport.ts
import { apiRequest } from "./core";

export interface CockpitEodSummary {
  headline: string;
  opened_today: number;
  closed_today: number;
  open_positions_now: number;
  alerts_needing_attention: number;
  lessons_available: number;
}

export interface CockpitEodPnl {
  realized_day: number | null;
  unrealized_snapshot: number | null;
  realized_basis: string;
  unrealized_basis: string;
}

export interface CockpitEodOpenPositionItem {
  asset_symbol: string;
  asset_id: string | null;
  asset_name: string | null;
  asset_detail_path: string | null;
  has_asset_context: boolean;
  side: string;
  qty: number | null;
  opened_at: string | null;
  unrealized_pnl: number | null;
}

export interface CockpitEodTradeItem {
  asset_symbol: string;
  asset_id: string | null;
  asset_name: string | null;
  asset_detail_path: string | null;
  has_asset_context: boolean;
  side: string;
  opened_at: string | null;
  closed_at: string | null;
  realized_pnl: number | null;
  close_reason: string | null;
}

export interface CockpitEodPaperActivity {
  opened_today: number;
  closed_today: number;
  current_open_positions: number;
}

export interface CockpitEodOpenPositions {
  count: number;
  items: CockpitEodOpenPositionItem[];
}

export interface CockpitEodClosedPositions {
  count: number;
  wins: number | null;
  losses: number | null;
  flat: number | null;
  unknown: number;
  best_trade: CockpitEodTradeItem | null;
  worst_trade: CockpitEodTradeItem | null;
  items: CockpitEodTradeItem[];
}

export interface CockpitEodIncidentItem {
  severity: string;
  code: string;
  title: string;
  source: string;
  created_at: string | null;
  detail: string | null;
}

export interface CockpitEodMonitorNote {
  title: string;
  detail: string;
  severity: string;
  created_at: string | null;
}

export interface CockpitEodLesson {
  title: string;
  detail: string;
  evidence_count: number;
}

export interface CockpitEodReportResponse {
  report_date: string;
  generated_at: string;
  mode: "paper";
  summary: CockpitEodSummary;
  paper_activity: CockpitEodPaperActivity;
  pnl: CockpitEodPnl;
  open_positions: CockpitEodOpenPositions;
  closed_positions: CockpitEodClosedPositions;
  alerts_or_incidents: CockpitEodIncidentItem[];
  monitor_notes: CockpitEodMonitorNote[];
  lessons: CockpitEodLesson[];
  recommended_actions: string[];
  limitations: string[];
}

export async function getCockpitEodReport(): Promise<CockpitEodReportResponse> {
  return apiRequest<CockpitEodReportResponse>("/cockpit/eod-report", {
    method: "GET",
  });
}